components/about/AboutMain.tsx

```diff
@@ -1,192 +1,57 @@
-  margin: -200px 0;
-  overflow: hidden;
-`;
-
-// overflow hidden used to mask photo
-const PhotoFrameHelper1 = styled.div`
-  position: absolute;
-  height: 800px;
-  top: -50%;
-  left: ${calcHexagonPointToBaseWidth(PHOTOHEXHEIGHT)}px;
-  overflow: hidden;
-  rotate: 30deg;
-`;
-
-// overflow hidden used to mask photo
-const PhotoFrameHelper2 = styled.div`
-  height: 800px;
-  width: ${PHOTOHEXHEIGHT}px;
-  overflow: hidden;
-  rotate: -60deg;
-`;
-
-// rotate and positioning fields in order to correct positioning changed by photo mask containers
-const Photo = styled.img`
-  position: absolute;
-  max-width: unset !important;
-  top: 25%;
-  left: -${calcHexagonPointToBaseWidth(PHOTOHEXHEIGHT)}px;
-  height: ${PHOTOHEXHEIGHT}px;
-  width: ${calcHexagonPointToPointWidth(PHOTOHEXHEIGHT)}px;
-  object-fit: cover;
-  rotate: 30deg;
-`;
-
 const AboutMain = () => (
-  <AboutMainContainer flexDirection='column' alignItems='center' fullWidth>
-    <RowContainer fullWidth>
-      <AboutItemContainer>
-        <SpinContainer>
-          <Hexagon />
-        </SpinContainer>
-        <AboutItemText>QUALITY</AboutItemText>
-      </AboutItemContainer>
+  <FlexContainer className='my-12 mx-0' flexDirection='column' alignItems='center' fullWidth>
+    <FlexContainer className='my-0 mx-100 first-of-type:mb-100 last-of-type:mt-100' fullWidth>
+      <div className='flex justify-center items-center h-200 w-[calc(200px/0.86602540378)]'>
+        <div className='absolute -z-1 spinAnimation gradie'>
+          <div className='hexagon' />
+        </div>
+        <div className='text-black text-shadowL shadow-white text-center w-[calc(200px/0.86602540378)]'>
+          QUALITY
+        </div>
+      </div>
       <FlexItem flex='1 0 0'/>
-      <AboutItemContainer>
-        <SpinContainer>
-          <Hexagon />
-        </SpinContainer>
-        <AboutItemText>DETAIL ORIENTED</AboutItemText>
-      </AboutItemContainer>
-    </RowContainer>
-    <PhotoFrame>
-      <PhotoFrameHelper1>
-        <PhotoFrameHelper2>
-            <Photo src='/images/profile.jpg'/>
-        </PhotoFrameHelper2>
-      </PhotoFrameHelper1>
-    </PhotoFrame>
-    <RowContainer fullWidth>
-      <AboutItemContainer>
-        <SpinContainer>
-          <Hexagon />
-        </SpinContainer>
-        <AboutItemText>COMMUNICATION</AboutItemText>
-      </AboutItemContainer>
+      <div className='flex justify-center items-center h-200 w-[calc(200px/0.86602540378)]'>
+        <div className='absolute -z-1 spinAnimation'>
+          <div className='hexagon' />
+        </div>
+        <div className='text-black text-shadowL shadow-white text-center w-[calc(200px/0.86602540378)]'>
+          DETAIL ORIENTED
+        </div>
+      </div>
+    </FlexContainer>
+    <div className='relative h-400 w-[calc(400px/0.86602540378)] -my-200 mx-0 overflow-hidden'>
+      <div className='absolute h-800 -top-1/2 left-[calc(((400px/0.86602540378)-400px)/2)] overflow-hidden rotate-30'>
```

Comment:
wtf is this?
left-[calc(((400px/0.86602540378)-400px)/2)]
tailwind sucks so ugly.

This is the most egregious problem i had. so before this was line 125 and the issue is that you can't do `w-[${calcHexagonPointToBaseWidth(PHOTOHEXHEIGHT)}px]`, because of something with the interpolation. I think it might be because the whole `${something}` is actually handled as a function call in runtime so it's too late for tailwind to create an arbitrary class but i have no clue.

anyway it's still ugly regardless but it would've been less so if i could've done that. so instead i had to hardcode a bunch of numbers and do the calculation right there. so it's definitely not pretty.

Reply:
https://tailwindcss.com/docs/content-configuration#dynamic-class-names